tap_asana/streams/projects.py

```diff
@@ -45,11 +46,34 @@ def get_objects(self):
     opt_fields = ",".join(self.fields)
     bookmark = self.get_bookmark()
     session_bookmark = bookmark
-    for workspace in self.call_api("workspaces"):
-      for project in self.call_api("projects", workspace=workspace["gid"], opt_fields=opt_fields):
-        session_bookmark = self.get_updated_session_bookmark(session_bookmark, project[self.replication_key])
-        if self.is_bookmark_old(project[self.replication_key]):
-          yield project
+
+    workspaces = self.fetch_workspaces()
```

Comment:
Can't you call `self.fetch_workspaces()` only single time and use it everywhere?